### PR TITLE
Variable explorer: Disregard list1 when sorting list1 against list2

### LINKED ISSUE #2995: Variable explorer - Sequence break with wrong mapping
the Name, Type, Size mapped wrong to each other after click each label name ( for order by), only the first python execution get the correct value. next time need to delete all content in variable explorer and re-run the code. 
please refer to the attached document for detail screen-shot.
[Spyder issue - Variable explorer .docx](https://github.com/spyder-ide/spyder/files/137574/Spyder.issue.-.Variable.explorer.docx)

### PR DESCRIPTION
**Pull request by jitseniesen**
Fixes #2995.

### CODE REVIEW

File: spyderlib/widgets/variableexplorer/utils.py [hunk 1]
```diff
@@ -217,7 +217,8 @@ def is_editable_type(value):
 def sort_against(lista, listb, reverse=False):
     """Arrange lista items in the same order as sorted(listb)"""
```
Comment (goanpeca) [inline spyderlib/widgets/variableexplorer/utils.py]: Can we fix this typo (I know you did not introduced it :-p ))?

**Comment by goanpeca** [comment]
@jitseniesen thanks for the help :-)

File: spyderlib/widgets/variableexplorer/utils.py (hunk 1 above)
Comment (ccordoba12) [inline spyderlib/widgets/variableexplorer/utils.py]: What's the typo?

Comment (goanpeca) [inline spyderlib/widgets/variableexplorer/utils.py]: lista ?

Comment (jitseniesen) [inline spyderlib/widgets/variableexplorer/utils.py]: I changed it, but I saw your comment too late to use your names ...

**Comment by ccordoba12** [comment]
Thanks a lot @jitseniesen, great work!! (as usual :-)

**Pull request merged**